### ISSUE TITLE
Support web3.js 1.2.2 or more than version

### ISSUE THREAD

**Issue by odanado** [ethereum-ts/TypeChain#187]
In web3.js 1.2.1, type definition files were divided into `@types/web3`. But, in 1.2.2 or more than version,  `web3`  have included type definition files.
ref: https://github.com/ethereum/web3.js/releases/tag/v1.2.2

Type definition files of `@types/web3`and type definition files of `web3@1.2.2` are not compatible. e.g. `web3/eth` to `web3-eth`.

Therefore, it is necessary to create a new target.

**Comment by krzkaczor** [comment]
@odanado thanks for reporting this! I think new target is not necessary - we should simply make current target work with latest stable 1.x version and in the readme note that for older web3 versions you can use this and that `typechain-target-web3v1` target.

BTW I really hope that web3 will start treating their TypeScript interfaces as public API and follow semver ie. don't break it on major. CC: @nivida

**Comment by odanado** [comment]
Thank you for comment. I think it's good to make a new version of `typechain-target-web3v1`.
I look forward to the new release :)

**Comment by krzkaczor** [comment]
@odanado I will try to do it over the weekend 👍

**Comment by krzkaczor** [comment]
It's released as `typechain-target-web3-v1@1.0.3`. @odanado  Let me know if you have any other problems 👍

**Comment by Velenir** [comment]
Types are still a bit off from web3.

1. Contract calls no longer return BN for `uint`, only 'string', so all `TransactionObject<BN>` should be changed to `TransactionObject<string>`
2. There's a ['connected' event](https://web3js.readthedocs.io/en/v1.2.4/web3-eth-contract.html#contract-events-return) for ContractEvent
```ts
export interface ContractEventEmitter<T> extends EventEmitter {
  on(event: 'connected', listener: (subscriptionId: string) => void): this
  ...
```

**Comment by nivida** [comment]
@Velenir Thanks for giving this hint! Version 1.0 of web3.js does return strings for all returned integers from a contract call who do overflow the 53bit of the JavaScript numbers type. 
We have planned to have the same behavior on the stable release of 2.0 because of compatibility reasons to other pure JS BigInt implementations as for example ``BigNumber.js``. If ``BN.js`` and ``BigNumber.js`` will add compatibility to BigInt in the meantime. Will we definitely check the possibility closer to return native BigInt's if supported by the current runtime.

**Comment by krzkaczor** [comment]
Released as `typechain-target-web3-v1@1.0.4`

**Comment by krzkaczor** [comment]
Thanks @Velenir for bringing this to my attention 👍
